move/axelar_gateway/sources/types/proof.move

```diff
@@ -66,19 +84,23 @@ public(package) fun recover_pub_key(
 /// Otherwise, the error `ELowSignaturesWeight` is raised.
 public(package) fun validate(
     self: &Proof,
-    message: vector<u8>
+    message: MessageToSign,
 ) {
     let signers = self.signers();
     let signatures = self.signatures();
     assert!(signatures.length() != 0, ELowSignaturesWeight);
 
     let threshold = signers.threshold();
     let mut total_weight: u128 = 0;
+    let mut signer_index = 0;
+    let message_bytes = bcs::to_bytes(&message);
 
     signatures.do_ref!(|signature| {
```

Comment:
i'd convert this back to a while loop. it's not very functional due to the mutation and the return in the middle, so i find it harder to read